ceph/rados/core_workflows.py

```diff
@@ -868,27 +868,29 @@ def change_osd_state(self, action: str, target: int) -> bool:
         Args:
             action: operation to be performed on the service, i.e start, stop, restart
             target: ID osd the target OSD
-        Returns: Pass -> True, Fail -> False
+        Returns: None
         """
-
         cluster_fsid = self.run_ceph_command(cmd="ceph fsid")["fsid"]
         osd_nodes = self.ceph_cluster.get_nodes(role="osd")
-        flag = False
-        for onode in osd_nodes:
-            res = self.collect_osd_daemon_ids(osd_node=onode)
-            if target in res:
-                cmd = f"systemctl {action} ceph-{cluster_fsid}@osd.{target}.service"
-                flag = True
-                log.info(
-                    f"Performing {action} on osd-{target} on host {onode.hostname}. Command {cmd}"
-                )
-                onode.exec_command(sudo=True, cmd=cmd)
-                # Sleeping for 5 seconds for status to be updated
-                time.sleep(5)
-                return True
-        if not flag:
-            log.error(f"osd-{target} not found on cluster")
+        all_daemons = self.run_ceph_command(cmd="ceph orch ps")
```

Comment:
Earlier implementation of ` change_osd_state ` made use of crush items/maps to locate the OSD host. which worked in most of the scenarios. ( made use of command ` sudo ceph osd ls-tree host ` to list all osd's under the host )

But with deployment of stretched clusters, the crush items are modified and the OSD's are moved to new crush items.
Hence made changes to fetch the osd daemon host details from ` ceph orch ps ` command, as this does not change with crush changes on cluster.

Reply:
Observed this issue with new suite : http://magna002.ceph.redhat.com/ceph-qe-logs/pawan/cephci-run-RT28OQ/